src/main/java/org/opentripplanner/routing/algorithm/strategies/SimpleEuclideanRWH.java

```diff
@@ -0,0 +1,103 @@
+package org.opentripplanner.routing.algorithm.strategies;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.core.vehicle_sharing.*;
+import org.opentripplanner.routing.edgetype.FreeEdge;
+import org.opentripplanner.routing.graph.Edge;
+import org.opentripplanner.routing.graph.Vertex;
+
+import java.util.Optional;
+import java.util.Set;
+
+import static java.lang.Double.min;
+import static org.opentripplanner.common.geometry.SphericalDistanceLibrary.fastDistance;
+
+public class SimpleEuclideanRWH implements RemainingWeightHeuristic {
+
+    private double lat;
+    private double lon;
+    private double bestMultiplier;
+
+    @Override
+    public void initialize(RoutingRequest options, long abortTime) {
+        calculateDestination(options.rctx.target);
+        calculateBestMultiplier(options);
+    }
+
+    private void calculateDestination(Vertex target) {
+        if (target.getDegreeIn() == 1) {
+            Optional<Edge> edge = target.getIncoming().stream().findFirst().filter(FreeEdge.class::isInstance);
+            if (edge.isPresent()) {
+                lat = edge.get().getFromVertex().getLat();
+                lon = edge.get().getFromVertex().getLon();
+                return;
+            }
+        }
+        lat = target.getLat();
+        lon = target.getLon();
+    }
+
+    private void calculateBestMultiplier(RoutingRequest options) {
+        bestMultiplier = options.routingReluctances.getWalkReluctance() / options.walkSpeed;
```

Comment:
Ochrona przed Div by Zero?

Reply:
może już jest, a może warto zrobić to po stronie options

Reply:
I added checking if values in request parameters are greater than zero